x-pack/plugins/ml/public/application/components/import_export_jobs/export_jobs_flyout/export_job_warning_callout.tsx

```diff
@@ -0,0 +1,176 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiCallOut, EuiText, EuiAccordion, EuiSpacer } from '@elastic/eui';
+import type { JobDependencies } from './jobs_export_service';
+
+interface Props {
+  jobs: JobDependencies;
+}
+
+export const ExportJobDependenciesWarningCallout: FC<Props> = ({ jobs: allJobs }) => {
+  const usingCalendars = allJobs.some((j) => j.calendarIds.length > 0);
+  const usingFilters = allJobs.some((j) => j.filterIds.length > 0);
+
+  const jobs = allJobs.filter(
+    ({ calendarIds, filterIds }) => calendarIds.length > 0 || filterIds.length > 0
+  );
+
+  if (usingCalendars === false && usingFilters === false) {
+    return null;
+  }
+
+  return (
+    <>
+      <EuiCallOut title={getTitle(jobs)} color="warning">
+        <FormattedMessage
+          id="xpack.ml.importExport.exportFlyout.exportJobDependenciesWarningCallout.calendarDependencies"
+          defaultMessage="When importing jobs into a new environment, calendars and filters which are relied upon by jobs will need to be recreated."
+        />
+        <EuiSpacer />
+
+        {usingCalendars && (
+          <EuiAccordion
+            id="advancedOptions"
+            paddingSize="s"
+            aria-label={i18n.translate(
+              'xpack.ml.importExport.exportFlyout.exportJobDependenciesWarningCallout.jobUsingCalendarsAria',
+              {
+                defaultMessage: 'Jobs using calendars',
+              }
+            )}
+            buttonContent={
+              <FormattedMessage
+                id="xpack.ml.importExport.exportFlyout.exportJobDependenciesWarningCallout.jobUsingCalendarsButton"
+                defaultMessage="Jobs using calendars"
+              />
+            }
+          >
+            <CalendarJobList jobs={jobs} />
+          </EuiAccordion>
+        )}
+
+        {usingFilters && (
+          <EuiAccordion
+            id="advancedOptions"
+            paddingSize="s"
+            aria-label={i18n.translate(
+              'xpack.ml.importExport.exportFlyout.exportJobDependenciesWarningCallout.jobUsingFiltersAria',
+              {
+                defaultMessage: 'Jobs using filters',
+              }
+            )}
+            buttonContent={
+              <FormattedMessage
+                id="xpack.ml.importExport.exportFlyout.exportJobDependenciesWarningCallout.jobUsingFiltersButton"
+                defaultMessage="Jobs using filters"
+              />
+            }
+          >
+            <FilterJobList jobs={jobs} />
+          </EuiAccordion>
+        )}
+      </EuiCallOut>
+
+      <EuiSpacer size="m" />
+    </>
+  );
+};
+
+const CalendarJobList: FC<{ jobs: JobDependencies }> = ({ jobs }) => {
+  return (
```

Comment:
Nit: this could be an implicit return like ({ jobs }) => (<> ...

Reply:
Fixed in https://github.com/elastic/kibana/pull/107416/commits/6b533f4b5adf3c972a291bd2c32333c7b2ebb2ea